src/pipeline-stages/src/main/scala/TextPreprocessor.scala

```diff
@@ -0,0 +1,141 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark
+
+import org.apache.spark.ml.Transformer
+import org.apache.spark.ml.param.{MapParam, Param, ParamMap}
+import org.apache.spark.ml.util.{ComplexParamsReadable, ComplexParamsWritable, Identifiable}
+import org.apache.spark.sql.types.{StringType, StructField, StructType}
+import org.apache.spark.sql.{DataFrame, Dataset}
+import org.apache.spark.sql.functions.udf
+import spray.json.DefaultJsonProtocol._
+
+class Trie(map: Map[Char, Trie] = Map.empty,
```

Comment:
Please check what the accessibility is by default.. make ``private[namespace]`` if necessary.

Reply:
Default is package level visibility. http://www.jesperdj.com/2016/01/08/scala-access-modifiers-and-qualifiers-in-detail/. That means I can leave as is correct?